tests/operator_test.go

```diff
@@ -234,8 +287,107 @@ var _ = Describe("Operator delete CDI tests", func() {
+			},
+			Spec: *localSpec,
+		}
+
+		_, err = f.CdiClient.CdiV1beta1().CDIs().Create(context.TODO(), cdi, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Recreating CDI operator")
+		_, err = f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Create(context.TODO(), cdiOperator, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verifying CDI apiserver, deployment, uploadproxy exist, before continuing")
+		Eventually(func() bool { return infraDeploymentAvailable(f, cr) }, CompletionTimeout, assertionPollInterval).Should(BeTrue(), "Timeout reading CDI deployments")
+
+		By("Testing all infra deployments have the chosen node placement")
+		for _, deploymentName := range []string{"cdi-apiserver", "cdi-deployment", "cdi-uploadproxy"} {
+			deployment, err := f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			match := tests.PodSpecHasTestNodePlacementValues(f, deployment.Spec.Template.Spec)
+			Expect(match).To(BeTrue(), fmt.Sprintf("node placement in pod spec\n%v\n differs from node placement values in CDI CR\n%v\n", deployment.Spec.Template.Spec, cr.Spec.Workloads))
+		}
+
+		By("Deleting modified CDI operator and CDI CR, original will be re-created")
+		err = f.CdiClient.CdiV1beta1().CDIs().Delete(context.TODO(), cr.Name, metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() bool { return infraDeploymentGone(f) && crGone(f, cr) }, 5*time.Minute, 2*time.Second).Should(BeTrue())
+
+		err = f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Delete(context.TODO(), "cdi-operator", metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+	})
 })
 
+func infraDeploymentAvailable(f *framework.Framework, cr *cdiv1.CDI) bool {
+	cdi, _ := f.CdiClient.CdiV1beta1().CDIs().Get(context.TODO(), cr.Name, metav1.GetOptions{})
+	for _, c := range cdi.Status.Conditions {
+		if c.Type == conditions.ConditionAvailable && c.Status != corev1.ConditionTrue {
+			return false
+		}
+	}
+
+	for _, deploymentName := range []string{"cdi-apiserver", "cdi-deployment", "cdi-uploadproxy"} {
+		_, err := f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func infraDeploymentGone(f *framework.Framework) bool {
+	for _, deploymentName := range []string{"cdi-apiserver", "cdi-deployment", "cdi-uploadproxy"} {
+		_, err := f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+		if !errors.IsNotFound(err) {
+			return false
+		}
+	}
+	return true
+}
+
+func crGone(f *framework.Framework, cr *cdiv1.CDI) bool {
+	_, err := f.CdiClient.CdiV1beta1().CDIs().Get(context.TODO(), cr.Name, metav1.GetOptions{})
+	if !errors.IsNotFound(err) {
+		return false
+	}
+	return true
+}
+
+func cdiOperatorGone(f *framework.Framework) bool {
```

Comment:
Please make sure to be explicit about if it is the deployment or the pod you are talking about. You can set the deployment replicas to 0 and have the pod removed that way and keep the deployment the same, or you can remove the deployment and create a new one with a different spec. It is unclear to me which is happening here (I think the second, but the naming is confusing)